lib/foreman_discovery/engine.rb

```diff
@@ -215,6 +216,9 @@ class Engine < ::Rails::Engine
       # Controller extensions
       ::HostsController.send :include, ForemanDiscovery::Concerns::HostsControllerExtensions
       ::Api::V2::FactValuesController.send :include, Api::V2::FactValuesControllerExtensions
+
+      # Graphql extensions
+      Types::Query.send(:include, Types::Extensions::ForemanDiscovery::QueryExtensions)
```

Comment:
I think you can use [register_graphql_query_field](https://github.com/theforeman/foreman/blob/ed3abb68ad71c365ab5f0992ec543234e1376a6a/app/registries/foreman/plugin.rb#L561-L563) 

```ruby
register_graphql_query_field : discovery_rule, 'Types::DiscoveryRule', :record_field
register_graphql_query_field : discovery_rules, 'Types::DiscoveryRule', :collection_field
```